saliency_metrics/metrics/insertion_deletion/insertion_deletion.py

```diff
@@ -0,0 +1,103 @@
+        if self.forward_batch_size <= 0:
+            raise ValueError(
+                f"""forward_batch_size should be greater than zero, but got
+                {self.forward_batch_size}."""
+            )
+        self.perturb_step_size = perturb_step_size
+        if self.perturb_step_size <= 0:
+            raise ValueError(
+                f"""perturb_step_size should be greater than zero and less than the number of elements in smap,
+                but got {self.perturb_step_size}."""
+            )
+
+    def evaluate(self, img: torch.Tensor, smap: torch.Tensor, target: int, **kwargs: Any) -> Dict:
+        num_pixels = torch.numel(smap)
+        if self.perturb_step_size >= num_pixels:
+            raise ValueError(
+                f"""perturb_step_size should be less than the number of elements in smap,
+                but got {self.perturb_step_size}."""
+            )
+
+        if img.dim() != 4 or img.size()[0] != 1:
+            raise ValueError(
+                f"""img should be a 4-D tensor with the size of the first dimension being 1,
+                             but got shape: {img.size()}."""
+            )
+
+        _, inds = torch.topk(smap.flatten(), num_pixels)
+        row_inds, col_inds = (torch.tensor(x) for x in np.unravel_index(inds.numpy(), smap.size()))
+
+        # deletion
+        del_perturbation = ProgressivePerturbation(img, torch.zeros_like(img), (row_inds, col_inds))
+        del_scores: List[float] = []
+        with torch.no_grad():
+            for batch in del_perturbation.perturb(
+                forward_batch_size=self.forward_batch_size, perturb_step_size=self.perturb_step_size
+            ):
+                scores = torch.softmax(self.classifier(batch), -1)[:, target]
+                del_scores.append(scores.cpu().numpy())
+
+        del_scores = np.concatenate(del_scores, 0)
+        del_auc = trapezoid(del_scores, dx=1.0 / len(del_scores))
+
+        blurred_img = self.gaussian_blur(img)
+        ins_perturbation = ProgressivePerturbation(blurred_img, img, (row_inds, col_inds))
+        ins_scores: List[float] = []
+        with torch.no_grad():
+            for batch in ins_perturbation.perturb(
+                forward_batch_size=self.forward_batch_size, perturb_step_size=self.perturb_step_size
+            ):
+                scores = torch.softmax(self.classifier(batch), -1)[:, target]
+                ins_scores.append(scores.cpu().numpy())
+
+        ins_scores = np.concatenate(ins_scores, 0)
+        ins_auc = trapezoid(ins_scores, dx=1.0 / len(ins_scores))
+
+        single_result = {
+            "del_scores": del_scores,
+            "ins_scores": ins_scores,
+            "img_path": kwargs["img_path"],
+            "del_auc": del_auc,
+            "ins_auc": ins_auc,
+        }
+        return single_result
+
+    def update(self, single_result: Dict) -> None:
+        self._result.add_single_result(single_result)
+
+    @property
```

Comment:
`get_result` is not a property in the base class. Please do not add the property decorator. Conventionally, a property will not be named as `get_xxx`.